.pre-commit-config.yaml

```diff
@@ -7,17 +7,14 @@ repos:
     -   id: mixed-line-ending
     -   id: end-of-file-fixer
         exclude: &exclude_pattern '^changelog.d/'
--   repo: https://github.com/pycqa/flake8
-    rev: 7.1.1
-    hooks:
-    -   id: flake8
-        name: "Flake8: critical"
-        args: ['--count', '--select=E9,F63,F7,F82', '--show-source', '--statistics']
-        exclude: smidumps/
```

Comment:
You are replacing a way to detect syntax errors with a linter that can reformat just about anything. You're not replacing like for like.

Reply:
Ruff *can* fix many of the linter errors it flags, but you need to explicitly ask for it to do so using the `--fix` option.  Reformatting is not enabled in this pipeline, nor should it be: It should really only be done manually by developer who wants it (or, perhaps, if using ruff to replace black, you would want it as a pre-commit hook).

I'm not sure which rules are enabled by default in ruff, but it believe it tries to stick closely to the same defaults as flake8.  

Since the old pipeline selected some specific flake8 linter rules, the new pipeline should probably select the same rules in ruff - and maybe all the other fluff of this PR would melt away (and would naturally belong in a separate PR)